plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -64,36 +64,36 @@ var defaultConfig = config.Configuration{
 	LatencyUnit:                 "ms",
 }
 
-func createTracingProvider() trace.TracerProvider {
-	exporter, err := stdout.New(stdout.WithPrettyPrint())
-	if err != nil {
-		panic(err)
-	}
+func tracerProvider(t *testing.T) (trace.TracerProvider, *tracetest.InMemoryExporter) {
+	exporter := tracetest.NewInMemoryExporter()
 	tp := sdktrace.NewTracerProvider(
 		sdktrace.WithSampler(sdktrace.AlwaysSample()),
 		sdktrace.WithBatcher(exporter),
 	)
-	return tp
+	defer tp.Shutdown(context.Background())
```

Comment:
defer fires when you exit this function. You're disabling provider even before you returned it.

Reply:
I tried testing this out without shutdown, but still failed to get span recorded.

Reply:
shutdown needs to be called from test functions
```
tp, exporter = tracerProvider(t)
defer tp.Shutdown()
```

Reply:
and you don't want `sdktrace.WithBatcher(exporter)`, you want to export immediately. Maybe `withSyncer`

Reply:
Ah, I see. I tried that option `WithSyncer`, but no chance. 
And If I change its argument to use `sdktrace`, `NewMetricsReader()` will have to be changed to support it. Right? But then, `otel.GetTr..()` doesn't support it.